float32 angle
bool direction
